### PR TITLE
Remove $options argument from Table::get().

### PR DESCRIPTION
**Pull request by ADmad**
Instead separate arguments are added for finder, cache and cacheKey. Variadic args added for using as query options.

### CODE REVIEW

File: src/ORM/Table.php [hunk 1]
```diff
@@ -1542,14 +1552,27 @@ public function get(mixed $primaryKey, array $options = []): EntityInterface
         }
         $conditions = array_combine($key, $primaryKey);
 
-        $cacheConfig = $options['cache'] ?? false;
-        $cacheKey = $options['key'] ?? false;
-        $finder = $options['finder'] ?? 'all';
-        unset($options['key'], $options['cache'], $options['finder']);
+        if (is_array($finder)) {
+            // deprecationWarning(
```
Comment (ADmad) [inline src/ORM/Table.php]: If the signature change is approved I'll uncomment this warning and update the tests.

File: src/Datasource/RepositoryInterface.php [hunk 2]
```diff
@@ -86,13 +87,23 @@ public function find(string $type = 'all', mixed ...$args): QueryInterface;
...
      * @return \Cake\Datasource\EntityInterface
      * @see \Cake\Datasource\RepositoryInterface::find()
      */
-    public function get(mixed $primaryKey, array $options = []): EntityInterface;
+    public function get(
+        mixed $primaryKey,
+        array|string $finder = 'all',
+        CacheInterface|string|null $cache = null,
+        Closure|string|null $cacheKey = null,
+        mixed ...$args
```
Comment (othercorey) [inline src/Datasource/RepositoryInterface.php:105]: Need to document $args

File: src/ORM/Table.php (hunk 1 above)
Comment (markstory) [reply src/ORM/Table.php]: I think the signature looks great here. We don't appear to be giving up any backwards compatibility and are getting typed parameters as well.

**Review by markstory — APPROVED** [review]

File: tests/TestCase/ORM/Association/BelongsToManyTest.php [hunk 3]
```diff
@@ -405,7 +405,7 @@ public function testFinderOption(): void
             'finder' => 'published',
         ]);
         $articles->updateAll(['published' => 'N'], ['id' => 1]);
-        $entity = $tags->get(1, ['contain' => 'Articles']);
+        $entity = $tags->get(1, ...['contain' => 'Articles']);
```
Comment (markstory) [inline tests/TestCase/ORM/Association/BelongsToManyTest.php:408]: Is this something we would want to do with rector? We could also use the rector rule I wrote for find() and apply it to get() as well.

Comment (ADmad) [reply tests/TestCase/ORM/Association/BelongsToManyTest.php:408]: Yes, the same rule for `find()` can work for `get()` too. Will just have to account for `key` in options array is now `$cacheKey` as argument name.

Comment (markstory) [reply tests/TestCase/ORM/Association/BelongsToManyTest.php:408]: Renaming the key is something we should be able to do with rector. I can put that together this week.

Comment (ADmad) [reply tests/TestCase/ORM/Association/BelongsToManyTest.php:408]: Thanks, this is ready to be merged then.

**Pull request merged**